python/lsst/meas/photocal/PhotoCal.py

```diff
@@ -401,64 +414,68 @@ def extractMagArrays(self, matches, filterName, keys):
+        refFluxArrList = [] # list of ref arrays, one per flux field
+        refFluxErrArrList = [] # list of ref flux arrays, one per flux field
+        for fluxField in fluxFieldList:
+            fluxKey = refSchema.find(fluxField).key
+            refFluxArr = np.array([m.first.get(fluxKey) for m in matches])
             try:
-                fluxErrKey = refSchema.find(flux + ".err").key
-                refFluxErr = np.array([m.first.get(fluxErrKey) for m in matches])
+                fluxErrKey = refSchema.find(fluxField + "Sigma").key
+                refFluxErrArr = np.array([m.first.get(fluxErrKey) for m in matches])
             except KeyError:
-                # Catalogue may not have flux uncertainties; HACK
+                # Reference catalogue may not have flux uncertainties; HACK
                 self.log.warn("Reference catalog does not have flux uncertainties for %s; using sqrt(flux)."
-                              % flux)
-                refFluxErr = np.sqrt(refFlux)
+                              % fluxField)
+                refFluxErrArr = np.sqrt(refFluxArr)
 
-            refFluxes.append(refFlux)
-            refFluxErrors.append(refFluxErr)
+            refFluxArrList.append(refFluxArr)
+            refFluxErrArrList.append(refFluxErrArr)
 
-        srcMag = -2.5*np.log10(srcFlux)
         if ct:                          # we have a colour term to worry about
-            refMag =  -2.5*np.log10(refFluxes[0]) # primary
-            refMag2 = -2.5*np.log10(refFluxes[1]) # secondary
+            refMagArr1 = np.array([abMagFromFlux(rf1) for rf1 in refFluxArrList[0]]) # primary
+            refMagArr2 = np.array([abMagFromFlux(rf2) for rf2 in refFluxArrList[1]]) # secondary
 
-            refMag = ct.transformMags(filterName, refMag, refMag2)
-            refFluxErr = ct.propagateFluxErrors(filterName, refFluxErrors[0], refFluxErrors[1])
+            refMagArr = ct.transformMags(filterName, refMagArr1, refMagArr2)
+            refFluxErrArr = ct.propagateFluxErrors(filterName, refFluxErrArrList[0], refFluxErrArrList[1])
         else:
-            refMag = -2.5*np.log10(refFluxes[0])
+            refMagArr = np.array([abMagFromFlux(rf) for rf in refFluxArrList[0]])
+
+        srcMagArr = np.array([abMagFromFlux(sf) for sf in srcFluxArr])
 
         # Fitting with error bars in both axes is hard, so transfer all
-        # the error to src, then convert to magnitude
-        fluxErr = np.hypot(srcFluxErr, refFluxErr)
-        magErr = fluxErr / (srcFlux * np.log(10))
+        # the error to src, then convert to magnitude;
+        # estimate the scaling first, to put ref errors in approximately the same system as src errors
+        meanSrcPerRef = afwMath.makeStatistics(srcFluxArr/refFluxArr, afwMath.MEANCLIP).getValue()
+        fluxErrArr = np.hypot(srcFluxErrArr, refFluxErrArr*meanSrcPerRef)
+        magErrArr = np.array([abMagErrFromFluxErr(fe, sf) for fe, sf in izip(fluxErrArr, srcFluxArr)])
```

Comment:
I have filed DM-2308 about this hack and the hack of using sqrt(refFlux) as the reference flux error if the error is unknown (though that is unlikely to be a problem for any reasonable photometric catalog).

Meanwhile I plan to adopt your suggestion of simply ignoring reference catalog errors until DM-2308 is dealt with. It is not ideal, but may be more robust than the current hacks.